src/components/profiles-modals/ImportProfileModal.vue

```diff
@@ -0,0 +1,480 @@
+    }
+
+    get activeProfileName(): string {
+        return this.$store.getters['profile/activeProfileName'];
+    }
+
+    get appName(): string {
+        return ManagerInformation.APP_NAME;
+    }
+
+    get profileList(): string[] {
+        return this.$store.state.profiles.profileList;
+    }
+
+    closeModal() {
+        this.newProfileName = '';
+        this.activeStep = 'IMPORT_UPDATE_SELECTION';
+        this.importUpdateSelection = null;
+        this.profileImportCode = '';
+        this.$store.commit('closeImportProfileModal');
+        this.listenerId = 0;
+    }
+
+    makeProfileNameSafe(nameToSanitize: string): string {
+        return sanitize(nameToSanitize);
+    }
+
+    async profileSelectOnChange(event: Event) {
+        if (event.target instanceof HTMLSelectElement) {
+            await this.$store.dispatch('profiles/setSelectedProfile', {profileName: event.target.value, prewarmCache: false});
+        }
+    }
+
+    doesProfileExist(nameToCheck: string): boolean {
+        if ((nameToCheck.match(new RegExp('^([a-zA-Z0-9])(\\s|[a-zA-Z0-9]|_|-|[.])*$'))) === null) {
+            return true;
+        }
+        const safe: string = this.makeProfileNameSafe(nameToCheck);
+        return (this.profileList.some(function (profile: string) {
+            return profile.toLowerCase() === safe.toLowerCase()
+        }));
+    }
+
+    // User selects importing via code from the UI.
+    async importProfileUsingCode() {
+        try {
+            const filepath = await ProfileImportExport.downloadProfileCode(this.profileImportCode.trim());
+            await this.importProfileHandler([filepath]);
+        } catch (e: any) {
+            const err = R2Error.fromThrownValue(e, 'Failed to import profile');
+            this.$store.commit('error/handleError', err);
+        }
+    }
+
+    // User selects importing from file from the UI.
+    importProfileFromFile() {
+        this.activeStep = 'FILE_SELECTION';
+        InteractionProvider.instance.selectFile({
+            title: 'Import Profile',
+            filters: [{
+                name: "*",
+                extensions: ["r2z", "r2x"]
+            }],
+            buttonLabel: 'Import'
+        }).then((value: string[]) => {
+            this.importProfileHandler(value);
+        })
+    }
+
+    // Handles the importing of the profile (doesn't matter if it is from file or code)
+    // When creating:
+    //      1. Reads and parses the mods to be installed
+    //      2. Creates an EventListener which gets triggered after the user has created a new profile. The callback then:
+    //          1. Downloads (and extracts) the mods to the profile
+    // When updating:
+    //      1. Reads and parses the mods to be installed
+    //      2. Creates an EventListener which gets triggered after the user has selected the old profile to be updated. The callback then:
+    //          1. Creates a temporary profile (called _profile_update)
```

Comment:
Actually to me it seems more like:

1. Deletes the temporary profile (called _profile_update) if it exists (leftover from previous failed run or something like that?)
2. Downloads (and extracts) the mods to temporary profile
3. Deletes the old profile (including the folder on disk)
4. Renames temporary profile to the chosen profile's name

Which makes more sense, since I assume the point of having the temp folder is to not lose the original profile if something goes wrong when downloading/extracting the mods.